src/environmentd/tests/server.rs

```diff
@@ -2054,6 +2054,78 @@ fn test_internal_console_redirect() {
     assert_eq!(redirected.load(Ordering::Relaxed), true);
 }
 
+#[mz_ore::test]
+#[cfg_attr(miri, ignore)] // too slow
+fn test_internal_http_auth() {
+    let server = util::start_server(util::Config::default()).unwrap();
+    let json = r#"{"query": "SELECT current_user;"}"#;
+    let json: serde_json::Value = serde_json::from_str(json).unwrap();
```

Comment:
you can use serde's `json!` macro to make this easier: `json!({"query": "SELECT current_user;"})`